uber/decorators.py

```diff
@@ -110,6 +110,42 @@ def charge(self, session, payment_id, stripeToken, stripeEmail='ignored', **igno
     return charge
 
 
+def cached(func):
+    func.cached = True
+    return func
+
+
+def cached_page(func):
+    from sideboard.lib import config as sideboard_config
+    innermost = get_innermost(func)
+    func.lock = RLock()
+    @wraps(func)
+    def with_caching(*args, **kwargs):
+        if hasattr(innermost, 'cached'):
+            fpath = os.path.join(sideboard_config['root'], 'data', func.__module__ + '.' + func.__name__)
+            with func.lock:
+                if not os.path.exists(fpath) or datetime.now().timestamp() - os.stat(fpath).st_mtime > 60 * 15:
+                    contents = func(*args, **kwargs)
+                    with open(fpath, 'wb') as f:
+                        f.write(contents)
+                with open(fpath, 'rb') as f:
+                    return f.read()
+        else:
+            return func(*args, **kwargs)
+    return with_caching
+
+
```

Comment:
CherryPy theoretically emits this info, so we could turn on CherryPy logging.  But unfortunately, CherryPy doesn't let us customized which pageloads we want logged, and we don't care about tracking our static files and really only care about tracking our dynamic pages.